src/Sarif/Readers/UriConverter.cs

```diff
@@ -26,12 +26,18 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
         {
             Uri uri = ((Uri)value);
 
-            if (uri.IsAbsoluteUri && uri.IsFile && (uri.LocalPath == uri.OriginalString))
+            string uriValue = UriToString(uri);
+
+            writer.WriteValue(uriValue);
+        }
+
+        internal static string UriToString(Uri uri)
+        {
+            if (uri.IsAbsoluteUri)
             {
-                uri = new Uri(uri.ToString());
+                return uri.AbsoluteUri;
             }
-
-            writer.WriteValue(uri.ToString());
+            return uri.ToString();
```

Comment:
This won't necessarily be escaped. See https://github.com/dotnet/roslyn/commit/c66413016d28dde42a1cd6cc669b3c4d47835deb